networking/src/APIClients/datamanagers/HostDataManagerAPIClient.ts

```diff
@@ -106,8 +149,9 @@ export class HostDataManagerAPIClient extends PlayDataManagerAPIClient {
     }
   };
 
-  private processAnswer(ans: any, teamAnswersQuestion: any, phase: string, teamName: string) {
+  private processAnswer(ans: any, teamAnswersQuestion: any, phase: IPhase, teamName: string) {
     const answerObj = this.createAnswerFromBackendData(ans.answer);
```

Comment:
I started running into a lot of issues when I tried to directly modify `this.teamAnswerQuestion` etc. I updated the whole class so that each function is immutable, making a copy of every object before I update it and then assigning the `this.` to it at the end. This made everything a bit more predictable